mmhuman3d/core/visualization/visualize_keypoints3d.py

```diff
@@ -158,8 +158,9 @@ def visualize_kp3d(
             allowed_suffix=['.mp4', '.gif', ''])
 
     # slice the frames
-    end = (min(end, num_frames - 1) + num_frames) % num_frames
-    kp3d = kp3d[start:end + 1]
+    end = (min(end, num_frames - 1) +
+           num_frames) % num_frames + 1 if end is not None else num_frames
```

Comment:
If we define end the same way as torch and numpy, I think we no longer need such a complicated manipulation of the `end`. For example, we do not have to consider the special case `-1` and it is handled in `kp3d[start:end]`.

The only necessary operation may be `end = num_frames if end is None else end`.

The same applies to other similar code fragments.

Reply:
already solved. Our `end` serves the same as `list`, `tensor`, and `numpy` slice.